src/main/src/main/java/org/geoserver/template/FeatureWrapper.java

```diff
@@ -347,16 +347,24 @@ public Set entrySet() {
                 entrySet = new LinkedHashSet<>();
                 final Collection<PropertyDescriptor> types = feature.getType().getDescriptors();
                 Name attName;
-                Map attributesMap;
                 for (PropertyDescriptor type : types) {
                     attName = type.getName();
-                    attributesMap = new AttributeMap(attName, feature);
-                    entrySet.add(
-                            new MapEntry<Object, Object>(attName.getLocalPart(), attributesMap));
+                    if (type.getMaxOccurs() > 1) {
+                        for (Property property : feature.getProperties(attName)) {
+                            addAttributeMap(attName, property);
```

Comment:
So this will cause the entryset to have multiple entries with the same key, effectively making this a multimap? However the rest of the map machinery won't work properly, e.g., if one goes and asks for a key directly (it's possible to use an attribute name directly too in the templates, see usage of "feature.NAME" or "feature.THUMBNAIL" [here](https://docs.geoserver.org/latest/en/user/tutorials/GetFeatureInfo/html.html#custom-templates)).

I'm guessing this works in the templates at hand, because they end up iterating the entry set directly, which is going to return the list of entries produced here.

Thinking out loud, this might need a different treatment, like storing the multiple values in a list instead.

Reply:
fixed